data/src/main/java/com/globant/data/repository/MemeRepositoryImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.globant.data.repository
+
+import com.globant.data.service.MemeServiceImpl
+import com.globant.domain.entity.Meme
+import com.globant.domain.entity.MemeDetail
+import com.globant.domain.repository.MemeRepository
+import com.globant.domain.util.NOT_FOUND
+import com.globant.domain.util.Result
+
+class MemeRepositoryImpl(private val memeService: MemeServiceImpl) : MemeRepository {
```

Comment:
Why this class is named **MemeRepositoryImpl** if you do not use a repository and it calls a service?

Reply:
This class has not to be here because I am not implementing the repository pattern. Thanks Mau by the explanation. Refactored!